.github/workflows/release.yaml

```
@@ -1,39 +1,114 @@
-name: release
+# This code is borrowed from https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/workflows/release.yaml
+name: Create Release
 
 on:
   push:
-    # Sequence of patterns matched against refs/tags
-    tags:
-    - "v*"
+    branches:
+      - main
+    paths:
+      - 'CHANGELOG/*.md'
 
-permissions: {}
+permissions:
+  contents: write # Allow to push a tag, create a release branch and publish a draft release.
 
 jobs:
-  build:
-    name: tag release
+  push_release_tags:
     runs-on: ubuntu-latest
-
-    permissions:
-      contents: write
-
-    if: github.repository == 'metal3-io/ip-address-manager'
+    outputs:
+      release_tag: ${{ steps.release-version.outputs.release_version }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # tag=v4.1.7
+        with:
+          fetch-depth: 0
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@e9772d140489982e0e3704fea5ee93d536f1e275 # tag=v45.0.1
+      - name: Get release version
+        id: release-version
+        run: |
+            if [[ ${{ steps.changed-files.outputs.all_changed_files_count }} != 1 ]]; then
+              echo "1 release notes file should be changed to create a release tag, found ${{ steps.changed-files.outputs.all_changed_files_count }}"
+              exit 1
+            fi
+            for changed_file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+              export RELEASE_VERSION=$(echo "${changed_file}" | grep -oP '(?<=/)[^/]+(?=\.md)')
+              echo "RELEASE_VERSION=$RELEASE_VERSION" >> $GITHUB_ENV
```

```suggestion
              echo "RELEASE_VERSION=${RELEASE_VERSION}" >> ${GITHUB_ENV}
```
Fix all the curlys in this file please.

---

fixed